### PR TITLE
feat: revamp Kubernetes menu

### LINKED ISSUE #7759: Kubernetes Navigation: Replace nav sections by group nav item + individual items
### Is your enhancement related to a problem? Please describe

As we grow the set of Kubernetes objects, there are too many to keep in the main nav, even with the current collapsable nav section. However, moving to a single item has the drawback of requiring two clicks to get to anything (feels like a second-class citizen). The current form will also conflict with hiding nav items as we plan in #6774. 

### Describe the solution you'd like

Nav sections should be removed/replaced. Each of the Kubernetes pages should be contributed as separate nav items, and the group should be contributed as a special 'group' nav item. Clicking on a group nav item opens a submenu identical to the look of the Settings or the AI Lab extension today, with a list of all the nav items in the group.

It is important to point out that both the group and each individual nav item are 'regular' nav items. In particular, they could each be added or removed from the main navigation individually as per #7758. Kubernetes would initially appear as just the one group nav item with a submenu, but here are some examples of what the user could do:
- Add Deployments and Services to the main nav, while keeping the Kubernetes group also visible (to get to all of the other items).
- Add Nodes to the main nav and hide the Kubernetes group, because they're only using nodes at the moment.
- Hide everything, because they're not using Kubernetes.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by feloy**
Signed-off-by: Philippe Martin <phmartin@redhat.com>

### What does this PR do?

Revamp Kubernetes menu according to #7759

TODO:

- [ ] Kubernetes Welcome page
- [ ] Pod page in Kubernetes section
- [ ] Hide Kubernetes resources menus by default

### Screenshot / video of UI

https://github.com/user-attachments/assets/4678ec95-671a-4a6c-ad7e-066f2ec7b67f

### What issues does this PR fix or reference?

Fixes #7759

### How to test this PR?

(TODO) 

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Comment by slemeur** [comment]
I don't think we should duplicate the items both in the navigation bar and in the sidebar. that looks confusing.

**Comment by deboer-tim** [comment]
Ditto, this looks a bit confusing. My initial intent was:

- Deployments, Nodes, Services, etc -> all still available nav menu items that open identically to today (full screen with no submenu), but hidden by default.
- Kubernetes -> new 'group' nav menu item that opens the submenu with all of the Kubernetes items. Clicking here opens then in the remaining space to the right. Visible by default.
- Current nav section removed.

In other words, when you start there is only the one Kubernetes nav option that looks/works like AI Lab, but if you use Deployments or Services frequently you can still add them to the main nav bar.

**Comment by cdrage** [comment]
IMHO my blunt feedback:
- Looks good but font needs to be WAYYY smaller / less left white space. Takes up a lot of information
- Pods should be shown so we don't go to another screen to access it, from call, should just load Pods section with filter set to Kubernetes only
- When clicking on deployments / pvc / etc on side bar, it should go "full screen" without side bar.

**Comment by feloy** [comment]
> IMHO my blunt feedback:
> 
> * When clicking on deployments / pvc / etc on side bar, it should go "full screen" without side bar.

Here is an overview with the latest PR of fullscreen for the Node resource. (code needs to be refactored). cc @cdrage @deboer-tim @axel7083 

https://github.com/user-attachments/assets/c626ce48-0978-4ea4-944d-6757415f7f9e

**Comment by cdrage** [comment]
> > IMHO my blunt feedback:
> > 
> > * When clicking on deployments / pvc / etc on side bar, it should go "full screen" without side bar.
> 
> Here is an overview with the latest PR of fullscreen for the Node resource. (code needs to be refactored). cc @cdrage @deboer-tim @axel7083
> 
>  nodes-fullscreen.mp4

From your video, when you click just on Nodes, Kubernetes is also still selected on the navbar. Nodes is also deselected when clicking on information on your nodes. I'd expect clicking on nodes it's "stay" selected if that makes sense.

**Comment by feloy** [comment]
> > > IMHO my blunt feedback:
> > > 
> > > * When clicking on deployments / pvc / etc on side bar, it should go "full screen" without side bar.
> > 
> > 
> > Here is an overview with the latest PR of fullscreen for the Node resource. (code needs to be refactored). cc @cdrage @deboer-tim @axel7083
> > nodes-fullscreen.mp4
> 
> From your video, when you click just on Nodes, Kubernetes is also still selected on the navbar. Nodes is also deselected when clicking on information on your nodes. I'd expect clicking on nodes it's "stay" selected if that makes sense.

Yes, I didn't work on this part yet. It is a quick overview to check if the Fullscreen is ok

**Comment by feloy** [comment]
Here a complete implementation of fullscreen:

https://github.com/user-attachments/assets/1a50e03d-cd1e-4462-bdda-95af911f3f07

**Comment by benoitf** [comment]
AFAIK fullscreen is a later one improvement

This PR should only focus on having the kubernetes submenu when we click on the left navbar and that's all)

**Comment by cdrage** [comment]
> Here a complete implementation of fullscreen:
> 
>  kube-fullscreen.mp4

Full screen and navbar looks great! Any plans to reduce the font like on the meeting in order to decrease the amount of whitespace the extra navbar now takes up?

**Comment by benoitf** [comment]
IMHO this PR is doing too many things at once. (hard to review)

Introducing the new menu is the only required feature. should only focus here (drop any icons from left menu sidebar)

it can be even 2 PR: one that bring the new menu and one that deletes left sidebar icons

Fullscreen or filtering is not part of this sprint. It's only part of a backlog list. (and AFAIK fullscreen would depend on the usage of the people.)

**Comment by benoitf** [comment]
@cdrage SettingsNavItem enhancement should not be part of this PR (here it's reusing the component that displays the sidebar)

make it smaller, collapsible, or any improvement etc should not be part of this PR

**Comment by slemeur** [comment]
> Fullscreen or filtering is not part of this sprint

+1, the initial ask is to make the navigation consistent with other parts of the application (settings, AI Lab...) 

We can create other issues (and add to https://github.com/containers/podman-desktop/issues/7762), we can even involve the community and start a discussion, asking people to share their opinion. @MariaLeonova and @ekidneyrh probably something you could look into.

**Comment by deboer-tim** [comment]
Just to be clear that they are not part of this issue, I've created new issues to track a Kubernetes Welcome page (#8825) and Pods page (#8826).

**Comment by benoitf** [comment]
@deboer-tim it doesn't duplicate https://github.com/containers/podman-desktop/issues/8818 and 
https://github.com/containers/podman-desktop/issues/8819 ?

**Comment by feloy** [comment]
closed in favor of #8833

**Comment by deboer-tim** [comment]
> @deboer-tim it doesn't duplicate #8818 and #8819 ?

🤦🏼‍♂️ yes, closed #8826.
